src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -272,36 +272,34 @@ export class MockHttpSocket extends MockSocket {
     // if it hasn't been flushed already (e.g. someone started reading request stream).
     this.flushWriteBuffer()
 
-    const httpHeaders: Array<Buffer> = []
+    // Create a `ServerResponse` instance to delegate HTTP message parsing,
+    // Transfer-Encoding, and other things to Node.js internals.
+    const serverResponse = new ServerResponse(new IncomingMessage(this))
 
-    httpHeaders.push(
-      Buffer.from(
-        `HTTP/1.1 ${response.status} ${
-          response.statusText || STATUS_CODES[response.status]
-        }\r\n`
-      )
+    /**
+     * Assign a mock socket instance to the server response to
+     * spy on the response chunk writes. Push the transformed response chunks
+     * to this `MockHttpSocket` instance to trigger the "data" event.
+     * @note Providing the same `MockSocket` instance when creating `ServerResponse`
+     * does not have the same effect.
+     * @see https://github.com/nodejs/node/blob/10099bb3f7fd97bb9dd9667188426866b3098e07/test/parallel/test-http-server-response-standalone.js#L32
+     */
+    serverResponse.assignSocket(
+      new MockSocket({
```

Comment:
I used the `MockSocket` we already have instead of `Writable`. While sockets extend streams, they implement additional methods on top. I know Node.js uses a simple `Writable` in their test but we have to acknowledge that test case is controlled. Our use cases are going to be _everything_ out there. Using an actual socket-compatible instance is crucial here.